inheritance.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+var Animal = function() {
+  this.stepsWalked = 0;
+  this.walk = function(steps) {
+    this.stepsWalked += steps;
+    console.log("i am walking ");
+  }
+  return this;
+};
+
+function Dog() {
+  Animal.call(this);
+}
+
+Dog.prototype = Object.create(Animal.prototype);
+Dog.prototype.constructor = Dog;
+
+Dog.prototype.wagTail = function() {
+  for (var i = 1; i <= 10; i++) {
```

Comment:
The idiomatic way to do numbering for for loops is from 0 to (max - 1), e.g. `for(var i = 0; i < 10; i++)`